File: Python Snippets/SumOfDigits.py
# Function that sums the digits of a integer using recursion
from utils import consolePrint
def SumOfDigits(n):
    assert n >= 0 and int(n) == n, "The number must be positive integer only!"
    if n < 10:
        return n
    elif 10 < n < 100  :
        return int(n/10) + n % 10
    else:
        return SumOfDigits(int(n/10)) + n%10

consolePrint("This module returns the sum of digits of a given positive integer.\nPlease enter a positive integer: ", "The sum of digits is ",SumOfDigits)

File: TypeScriptModules/sumOfDigits.ts
import launchConsole from './utils/utils.ts'

const sumOfDigits = (n: number): number => {
  if (n < 10) return n
  else if (n > 10 && n < 100) return Math.floor(n / 10) + (n % 10)
  else return sumOfDigits(Math.floor(n / 10)) + (n % 10)
}

launchConsole(
  'This module returns the sum of digits of a given positive integer.\nPlease enter a positive integer: ',
  'The sum of digits is',
  sumOfDigits
)

export default sumOfDigits
